### ISSUE TITLE
[Feature] Refresh root folder on change of contents

### ISSUE THREAD

**Issue by chip33** [Hofknecht/SystemTrayMenu#330]
As the subfolders automatically show a change of contents it would be good if the root folder could also refresh automatically if the contents are changed.

A file watcher could be used to monitor the root folder for changed contents and perhaps the refresh button would no longer be needed

**Comment by Hofknecht** [comment]
@chip33 
Thank you for the great feedback!

this is still one of our very big milestones, we've always pushed the idea back, but it's a very important feature we should tackle it soon.
It consists of several parts. One is the the filewatcher, we only had a few concerns about whether it always works, e.g. in a network path or missing permissions. Second, then trigger that, here we can proceed as with the non-cached menu, so to speak simply set a flag that it now has to be refreshed once the next time it is opened. and finally the supreme discipline that we do live refresh, we will need that later, e.g. to move files from one folder to another with live refresh

**Comment by Hofknecht** [comment]
@chip33 
Finally we made it. It took a while, but it works great now. We did a major redesign in many places.
It also works for everything added to the main menu via additional folders. Even with icon or attribute changes. The width of the menu also adapts.
(Theoretically, we could do the same for the subfolders, but they will be renewed anyway if you select the folder again.)
I hope no mistakes have crept in. We tested a lot and fixed all the new bugs we found. We'll check that out a bit more.

So it is no longer necessary to do a restart application when files/folder added/removed or changes are made!
